staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer.go

```diff
@@ -138,20 +153,79 @@ func NewFilteredDynamicInformer(client dynamic.Interface, gvr schema.GroupVersio
 			resyncPeriod,
 			indexers,
 		),
+		stopCh:  stopCh,
+		factory: factory,
 	}
 }
 
 type dynamicInformer struct {
 	informer cache.SharedIndexInformer
 	gvr      schema.GroupVersionResource
+	stopCh   chan struct{}
+	factory  *dynamicSharedInformerFactory
 }
 
 var _ informers.GenericInformer = &dynamicInformer{}
 
 func (d *dynamicInformer) Informer() cache.SharedIndexInformer {
-	return d.informer
+	return d
 }
 
 func (d *dynamicInformer) Lister() cache.GenericLister {
 	return dynamiclister.NewRuntimeObjectShim(dynamiclister.New(d.informer.GetIndexer(), d.gvr))
 }
+
+var _ cache.SharedIndexInformer = &dynamicInformer{}
+
+// It would be much cleaner if we could embed the struct cache.sharedIndexInformer in struct dynamicInformer
+// However, since cache.sharedIndexInformer is not exported, we have to define the delegation methods below.
+func (d *dynamicInformer) AddEventHandler(handler cache.ResourceEventHandler) {
+	d.informer.AddEventHandler(handler)
+}
+func (d *dynamicInformer) AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration) {
+	d.informer.AddEventHandlerWithResyncPeriod(handler, resyncPeriod)
+}
+func (d *dynamicInformer) GetStore() cache.Store {
+	return d.informer.GetStore()
+}
+func (d *dynamicInformer) GetController() cache.Controller {
+	return &dummyController{informer: d}
+}
+
+// Use local stopCh, the passed in stopCh is ignored.
+// The informer is stopped via its controller
+func (d *dynamicInformer) Run(stopCh <-chan struct{}) {
+	d.informer.Run(d.stopCh)
+}
+func (d *dynamicInformer) HasSynced() bool {
+	return d.informer.HasSynced()
+}
+func (d *dynamicInformer) LastSyncResourceVersion() string {
+	return d.informer.LastSyncResourceVersion()
+}
+func (d *dynamicInformer) AddIndexers(indexers cache.Indexers) error {
+	return d.informer.AddIndexers(indexers)
+}
+func (d *dynamicInformer) GetIndexer() cache.Indexer {
+	return d.informer.GetIndexer()
+}
+
+type dummyController struct {
+	informer *dynamicInformer
+}
+
+func (v *dummyController) Run(stopCh <-chan struct{}) {
+	<-stopCh
```

Comment:
Can we just make the caller call factory.RemoveResource() directly, and let RemoveResource() stop the informer. I think it's better than letting caller manage individual stop channels for every resource.